pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/metrics/impl/xml/CycloTest.xml

```diff
@@ -389,6 +389,71 @@ class Foo {
         for (int i = 0; i != import1Tokens.length && i != import2Tokens.length; i++) {
         }
     }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Multiple switch labels should count each</description>
+        <expected-problems>2</expected-problems>
+        <expected-messages>
+            <message>'Foo#bar()' has value 1.</message>
+            <message>'Foo#foo()' has value 3.</message>
+        </expected-messages>
+        <code><![CDATA[
+class Foo {
+    void bar(){}
+    void foo() {
+        int x=0;
+        switch (x) {                 // +1
+            case 1, 2: foo(); break; // +2
+            default: bar(); break;
+        }
+    }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Switch labeled blocks should count</description>
+        <expected-problems>2</expected-problems>
+        <expected-messages>
+            <message>'Foo#bar()' has value 1.</message>
+            <message>'Foo#foo()' has value 5.</message>
+        </expected-messages>
+        <code><![CDATA[
+class Foo {
+    void bar(){}
+    void foo() {
+        int x=0;
+        boolean a, b;
+        switch (x) {                            // +1
```

Comment:
Just being picky, the +1 is not for the switch, but for the method block

```suggestion
        switch (x) {
```

Reply:
I think, being picky here, is extremely useful :) This hopefully avoids later questions.... It's important, that the test case is correct (including the comments). So, the "+1" should be added to line 427 then?